natlas-server/app/instrumentation/__init__.py

```diff
@@ -0,0 +1,50 @@
+import flask
+import sentry_sdk
+import threading
+from opencensus.ext.ocagent import stats_exporter as ocagent_stats_exporter
+from opencensus.ext.ocagent import trace_exporter as ocagent_trace_exporter
+from opencensus.ext.flask.flask_middleware import FlaskMiddleware
+from opencensus.trace.print_exporter import PrintExporter
+from opencensus.trace import config_integration, samplers
+from opencensus.trace import tracer as tracer_module
+from opencensus.trace import execution_context
+from .sentryio_middleware import SentryIoContextMiddleware
+from urllib.parse import urlparse
+
+
+SERVICE_NAME = 'natlas-server'
+template_span = threading.local()
+
+
+def render_template_end(*kargs, **kwargs): # sender, template, context):
+	span = template_span.x
+	template_span.x = None
+	span.__exit__(None, None, None)
+
+
+def render_template_start(app, template, context):
+	tracer = execution_context.get_opencensus_tracer()
+	template_span.x = tracer.span(name="render_template")
+	template_span.x.add_attribute('flask.template', template.name)
+
+
+def initialize_opencensus(config, flask_app):
+	if config.opencensus_enable:
+		agent = config.opencensus_agent
+		print("OpenCensus enabled and reporting to %s using gRPC" % agent)
```

Comment:
Instead of using the `%` technique for string formatting, I've been trying to move towards using f-strings where possible, and `.format()` with explicitly named variables elsewhere.

The f-string equivalent of this line is:

```
print(f"OpenCensus enabled and reporting to {agent} using gRPC")
```

and the `.format()` equivalent using named variables would be:

```
print("OpenCensus enabled and reporting to {agent} using gRPC".format(agent=agent))
```